src/traces/sankey/index.js

```diff
@@ -0,0 +1,30 @@
+/**
+* Copyright 2012-2017, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var Plot = {};
+
+Plot.attributes = require('./attributes');
+Plot.supplyDefaults = require('./defaults');
+Plot.calc = require('./calc');
+Plot.plot = require('./plot');
+
+Plot.moduleType = 'trace';
+Plot.name = 'sankey';
+Plot.basePlotModule = require('./base_plot');
+Plot.categories = [];
```

Comment:
Is trace-wide `opacity` a thing for sankey traces? If not, `categories` should include the `noOpacity` flag like we did for parcoords in https://github.com/plotly/plotly.js/pull/1506